### PR TITLE
Update Kitchen Suites

### PR DESCRIPTION
**Pull request by damacus**
- Adds dokken
- Adds Amazonlinux-2
- Removes Fedora
- Remove references to Ubuntu 14.04

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a <a href="https://github.com/sous-chefs/openvpn/blob/master/CHANGELOG.md">CHANGELOG</a> entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: .kitchen.yml [hunk 1]
```diff
@@ -1,59 +1,29 @@
 ---
 driver:
-  name: dokken
-  privileged: true  # because Docker and SystemD/Upstart
-  chef_version: <%= ENV['CHEF_VERSION'] || 'current' %>
-
-transport:
-  name: dokken
-  require_chef_omnibus: <%= ENV['CHEF_VERSION'] || 'latest' %>
+  name: vagrant
```
Comment (jeffbyrnes) [inline .kitchen.yml]: So… I’m confused.

As of d1e970d0b877a23343249d15026393e1ec2651bd (the commit just before #143 landed) Kitchen was only using dokken, and vagrant had been dropped.

I’m good if we want both, though it seems counterproductive to keep going back & forth on this with every pull request.

My preference is to only use dokken, however, as it’s simpler and means that local testing mirrors what the automated testing does.

File: recipes/service.rb [hunk 2]
```diff
@@ -29,6 +29,15 @@
...
   end
+when 'amazon'
+  if node['platform_version'] >= '2'
+    link "/etc/systemd/system/multi-user.target.wants/openvpn@#{node['openvpn']['type']}.service" do
+      to '/usr/lib/systemd/system/openvpn@.service'
+    end
+    service_name = "openvpn@#{node['openvpn']['type']}.service"
+  else
+    service_name = 'openvpn'
+  end
```
Comment (jeffbyrnes) [inline recipes/service.rb]: 🎉

File: .circleci/config.yml [hunk 3]
```diff
@@ -11,35 +12,9 @@ workflows:
...
-          requires: [danger, lint]
-      - kitchen/dokken-single:
-          name: server-ubuntu-1604
-          suite: server
-          platform: ubuntu-1604
-          requires: [danger, lint]
-      - kitchen/dokken-single:
-          name: server-ubuntu-1804
+      - kitchen/dokken:
+          name: server
```
Comment (jeffbyrnes) [inline .circleci/config.yml]: Ah, much nicer than all that boilerplate.

File: .kitchen.yml (hunk 1 above)
Comment (damacus) [reply .kitchen.yml]: We're using dokken everywhere else, and I'd like to stick to this pattern. Giving people the choice to run it in a proper vm or docker with slimmed down network options is always useful.

File: recipes/service.rb (hunk 2 above)
Comment (damacus) [reply recipes/service.rb]: It's ugly and it, almost works! Just some annoying docker networking on amazon to work out now

File: .circleci/config.yml (hunk 3 above)
Comment (damacus) [reply .circleci/config.yml]: Though it does run it in serial, so it's a bit slower.

File: .kitchen.yml (hunk 1 above)
Comment (Xorima) [reply .kitchen.yml]: 👍 Dokken doesn't work on windows ... so I need this in order to test lol

File: .kitchen.dokken.yml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ---
 driver:
   name: dokken
-  privileged: true
+  privileged: false
```
Comment (Xorima) [inline .kitchen.dokken.yml]: can't we just remove this line? Isn't it default

Comment (damacus) [reply .kitchen.dokken.yml]: Just testing some stuff, to do with /sbin/init and agetty

Comment (damacus) [reply .kitchen.dokken.yml]: Seemingly the CentOS-6 builds are hitting this issue

https://github.com/moby/moby/issues/4040

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,7 +1,8 @@
+---
 version: 2.1
 
 orbs:
-  kitchen: sous-chefs/kitchen@1.0.0
+  kitchen: sous-chefs/kitchen@1.0.1
```
Comment (Xorima) [inline .circleci/config.yml]: should we just bring this up to 2 now?

Comment (damacus) [reply .circleci/config.yml]: Yeah

**Comment by damacus** [comment]
This needs starting from scratch again!

**Pull request closed without merging**